vpnbook

```diff
@@ -75,86 +77,144 @@ __check_requirements(){
-					OPENVPN_EXIT_STATUS=$?
-					if [ ${OPENVPN_EXIT_STATUS} -eq 0  ]; then
-						return 0
-					fi
-				fi
-			done
-		fi
-		return 1
-	else 
-		echo "Method called with an empty input parameter..."
-		return 1
-	fi
+        # Capture the process ID of the OpenVPN process
+        vpn_pid=$!
+
+        # Wait for a few seconds to allow OpenVPN to attempt the connection
+        sleep 10
+
+        # Check if the process is still running and check the log for AUTH_FAILED
+        if grep -q "AUTH_FAILED" ./tmp/vpnbook.log; then
+            echo "AUTH_FAILED detected. Please enter the password manually(check https://www.vpnbook.com/freevpn):"
+            read new_pwd  # -s makes sure the password is not displayed
+
+            # Update the login.conf with the new password
+            echo "vpnbook" > "${vpnbook_folder}/login.conf"
+            echo "${new_pwd}" >> "${vpnbook_folder}/login.conf"
+
+            # Kill the previous OpenVPN process
+            sudo kill "$vpn_pid"
+
+            # Retry the connection with the new password
+            echo "Retrying connection with the new password..."
+            sudo openvpn --config "${vpnbook_folder}/${server}/vpnbook-${l_server}-tcp443.ovpn" \
+                         --auth-user-pass "${vpnbook_folder}/login.conf" \
+                         --pull --route-ipv6 2000::/3 \
+                         --dhcp-option DNS6 2001:4860:4860::8888 > ./tmp/vpnbook.log 2>&1 &
+            
+            vpn_pid=$!  # Update the PID of the new process
+            sleep 10    # Give time for the connection attempt
+        fi
+
+        # Check if OpenVPN is still running
+        if ps -p "$vpn_pid" > /dev/null; then
+            echo "VPN connection established successfully."
+        else
+            echo "VPN connection failed. Please check the log at ./tmp/vpnbook.log for more details."
+        fi
+
+    else
+        echo "VPN configuration file not found: ${vpnbook_folder}/${server}/vpnbook-${l_server}-tcp443.ovpn"
+        return 1
+    fi
 }
+__connect_to_first_available(){
+    local server=$1
+    local OPENVPN_EXIT_STATUS
+    
+    if [ "${server}" != "" ]; then
+       
+        echo "Trying to Establish the Vpn Tunnel to ${server}..."
+        __connect_to ${server}
+        OPENVPN_EXIT_STATUS=$?
+
+        if [ ${OPENVPN_EXIT_STATUS} -eq 0 ]; then
+            echo "Vpn Tunnel established to ${server}"
 
+           
+            __check_if_up ${vpnbook_if_name}
+            if [ $? -eq 1 ]; then
+                echo "VPN interface is active"
+                return 0  # Success
+            else
+                echo "VPN interface ${vpnbook_if_name} is not active"
+                return 1  # Failure, interface didn't come up
+            fi
+         
+        else
+            echo "Failed to establish Vpn Tunnel to ${server}"
+        fi
+    else
+        echo "No server specified. Trying default servers..."
+       
+        for i in pl134 de20 us1 uk205 ca149 fr200; do
```

Comment:
What is your take if we rename this variable to "l_server" or "server_i"?

Reply:
Sorry Didnt understood that

Reply:
This variable is named _i_ which works great for loops with integers but in this case we iterate over a list of server names. My suggestion is to use a more suitable name like "server_i" or "l_server" for this variable.

P.S: I am not a fan of prefixes to discriminate between "global" variables and "local" variables (e.g. l_) so I would rather use "server_i" but given that this script has no real coding convention I don't mind using "l_server".

Reply:
Did that